File: ServiceAPI/Entities/Student.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using ServiceAPI.Entities;
using System;
using System.Collections.Generic;

namespace ServiceAPI.Dal
{
    public class Student
    {
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }

        public string Name { get; set; }
        public string Surname { get; set; }
        public string Email { get; set; }
        public DateTime DateOfBirth { get; set; }
        public IList<string> Courses { get; set; } //List of course IDs useful to find all courses to which student is enrolled in db
        public IList<TestPrenotation> Prenotations { get; set; } //List of Test Prenotations --> Exams to be faced by student
        public IList<ExamRegistration> Registrations { get; set; } //List of passed and recorded exams
    }
}

File: GestioneMaterie/gestione_materie/ClientApp/app/components/students/student.ts
import{Registration} from './registration';
import {Prenotation} from './prenotation';

export interface Student {
    id: string;
    name: string;
    surname: string;
    email: string;
    dateOfBirth: Date;
    courses: string;
    registrations: Registration[];
    prenotations: Prenotation[];
    
}